crates/net/network/src/network.rs

```diff
@@ -203,6 +203,11 @@ impl NetworkHandle {
     pub fn bandwidth_meter(&self) -> &BandwidthMeter {
         &self.inner.bandwidth_meter
     }
+
+    /// Send message to gracefully shutdown node.
+    pub fn shutdown(&self) {
+        self.send_message(NetworkHandleMessage::Shutdown);
```

Comment:
this should be an async function, see comment here https://github.com/paradigmxyz/reth/issues/812#issuecomment-1401049144

Reply:
what's the comment? not sure why it needs to be async myself

Reply:
it needs to wait until the shutdown was completed (waiting for response)